hack/build-with-blog.sh

```diff
@@ -1,53 +1,111 @@
+  pushd "temp/docs-$latest"
+  KNATIVE_VERSION=$latest mkdocs build -d ../../site/docs
+  popd
+
+  # Previous release branches release-$version to /v$version-docs
+  versionjson=""
+  for i in "${!previous[@]}"; do
+    version=${previous[$i]}
+    versionjson+="{\"version\": \"v$version-docs\", \"title\": \"v$version\", \"aliases\": [\"\"]},"
+    git clone --depth 1 -b ${BRANCHES[$i+1]} https://github.com/${REPOS[i+1]}/docs "temp/docs-$version"
+    pushd "temp/docs-$version"
+    KNATIVE_VERSION=$version VERSION_WARNING=true mkdocs build -d "../../site/v$version-docs"
+    popd
+  done
+
+  # Set up the version file to point to the built docs.
+  cat << EOF > site/versions.json
+  [
+    {"version": "docs", "title": "v$latest", "aliases": [""]},
+    $versionjson
+    {"version": "development", "title": "(Pre-release)", "aliases": [""]}
+  ]
+EOF
+fi
+
+if [ -z "$SKIP_BLOG" ]; then
+  # Clone out the website and community repos for the hugo bits.
+  # TODO(jz) Cache this and just do a pull/update/use siblings for local dev flow.
+  git clone --depth 1 https://github.com/knative/website temp/website
+  pushd temp/website; git submodule update --init --recursive --depth 1; popd
+  git clone -b ${community_branch} --depth 1 https://github.com/${community_repo}/community temp/community
+
+  # Move blog files into position
+  mkdir -p temp/website/content/en
+  cp -r blog temp/website/content/en/
+
+  # Clone community/ in to position too
+  # This is pretty weird: the base community is in docs, but then the
+  # community repo is overlayed into the community/contributing subdir.
+  cp -r community temp/website/content/en/
+  cp -r temp/community/* temp/website/content/en/community/contributing/
+  rm -r temp/website/content/en/community/contributing/elections/2021-TOC # Temp fix for markdown that confuses hugo.
+
+  # See https://github.com/knative/website/blob/main/scripts/processsourcefiles.sh#L125
+  # For the reasoning behind all this.
+  echo 'Converting all links in GitHub source files to Hugo supported relative links...'
+  pushd temp/website
+  # Convert relative links to support Hugo
+  find . -type f -path '*/content/*.md' ! -name '*_index.md' ! -name '*index.md' ! -name '*README.md' \
+    ! -name '*serving-api.md' ! -name '*eventing-contrib-api.md' ! -name '*eventing-api.md' \
+    ! -name '*build-api.md' ! -name '*.git*' ! -path '*/.github/*' ! -path '*/hack/*' \
+    ! -path '*/node_modules/*' ! -path '*/test/*' ! -path '*/themes/*' ! -path '*/vendor/*' \
+    -exec sed -i '/](/ { s#(\.\.\/#(../../#g; s#(\.\/#(../#g; }' {} +
+  # Convert all relative links from README.md to index.html
+  find . -type f -path '*/content/*.md' ! -name '_index.md' \
+      -exec sed -i '/](/ { /http/ !{s#README\.md#index.html#g} }' {} +
+  # Convert all Markdown links to HTML
+  find . -type f -path '*/content/*.md' \
+      -exec sed -i '/](/ { /http/ !{s#\.md##g} }' {} +
+
+  # Run the hugo build as normal!
+  npx hugo
```

Comment:
```suggestion
hugo
```
I changed but know postcss not working
```
+ hugo
Start building sites … 
Total in 707 ms
Error: Error building site: POSTCSS: failed to transform "scss/main.css" (text/css). Check your PostCSS installation; install with "npm install postcss-cli". See https://gohugo.io/hugo-pipes/postcss/
```
maybe we should add to the path `export PATH="$PWD/node_modules/.bin:$PATH"`

Reply:
oh yes that's what we already have
```
 Run the hugo build as normal!

# need postcss cli in PATH
PATH=${PATH}:${PWD}/node_modules/.bin
pushd temp/website
hugo
popd
```

Reply:
yep with worked with 
```
# need postcss cli in PATH
PATH=${PATH}:${PWD}/node_modules/.bin
hugo
popd
```

Reply:
I think `export PATH="$PWD/node_modules/.bin:$PATH"` is exactly what npx does...

Reply:
hmm hmm hmm wonder why npx doesnt work but this does 🤔 maybe my mental model of what npx does is wrong

Reply:
doesnt really matter, Ill just do the PATH hack for now

Reply:
maybe you were thinking on implementing `npm run hugo` and have a script entry in package.json?

Reply:
🤷 🤷 🤷  no idea, but I pushed a version with the PATH set in an env var instead